### DISCUSSION TITLE
_nuxt folder cannot be found on sub directory domain

### DISCUSSION THREAD

**Discussion by NLDylan** [nuxt/nuxt#16926] [Questions]
hi!

I am currently trying to deploy a test environment on digital ocean app platform with Nuxt 3 apps.
-> Example.com - Container 1
-> Example.com/dashboard - Container 2

Container 1 deploys fine and container 2 has issues with the /_nuxt folder.
I get the following errors:
Cannot find static asset /_nuxt/asyncData-5b894112.mjs
Cannot find static asset /_nuxt/mijn-profiel-e6216823.mjs
Cannot find static asset /_nuxt/entry.39106ec7.css

When i look in the console i see that container 2 is trying to access the _nuxt folder on the root domain.
While its living in the subdirectory example.com/dashboard.
If i place dashboard/_nuxt as the url then the assets and files are loaded.

How can i fix this by adding dashboard to my /_nuxt url structure?

Greetings,

Dylan

**Comment by danielroe** [comment]
You can set `app.baseURL` in your Nuxt config, or set the NUXT_APP_BASE_URL environment variable at runtime.

**Reply by NLDylan** [reply]
Hi @danielroe,

Cannot really get this to work with app.baseURL.

What i have done:
1. Docker container that listens to http route example.com/dashboard
2. Add router: { base: '/dashoard/' } to nuxt.config.ts
3. added NUXT_APP_BASE_URL but then pages are not found.

The nuxt 3 application that needs to listen to the example.com/dashboard route is living in the root of the nuxt project.
Is this something that needs to change?

If i just do the first two steps i do see the url and the html. it just cannot find the /dashboard/_nuxt/ folder with the created files. its actually looking at the /_nuxt/ path.

https://github.com/nuxt/nuxt.js/issues/2021
Is actually the same problem but he is talking about that hes project folder has a SUBDIR called DOCS.

![Schermafbeelding 2022-09-04 om 18 35 52](https://user-images.githubusercontent.com/61246512/188323849-b3479ffb-c7f9-4459-a260-0c59cf2ccead.png)

I have seen multiple suggestions:
- router.base
- Public.path
- .app.baseURL

Any feedback on where i need to look?

Greetings,

Dylan

**Reply by danielroe** [reply]
You should not set router.base. If you provide a minimal reproduction I will check it out.

**Reply by NLDylan** [reply]
https://gitlab.com/NLDylan/test
This is the same structure i have on my project. It only has a clean nuxt.config.ts file with no router.base config added.
What do i need to do to this repo to host this on example.com/dashboard -> example.com/dashboard/toernooien would be a route added in this project.

The container in Digital ocean listens to http route domainname/dashboard which works, but somehow nuxt is not being able to find the /_nuxt folder as explained above.

Greetings,

Dylan

**Reply by NLDylan** [reply]
@danielroe Dont waste any time just yet by investigating something.
I think i have found something that is why this is not working, but i need some time to test that out.
Il report back.

**Comment by Flanker72** [comment]
@NLDylan I have the very similar problem launching Nuxt under the Express server to achieve multiple themes being served from one entry point.
You may have a look at my solution, it's rather dirty, however it works: https://github.com/VirtoCommerce/vc-theme-b2b-nuxt/blob/rnd/multitheme/index.mjs
("/mercury" route isn't working since there was some apis changed).

Key points are:
- set app.baseUrl in the nested theme's nuxt.config
- override the url to make the routing works properly

**Reply by NLDylan** [reply]
@Flanker72 the link is not working as expected.

@danielroe i have created a test environment with the repo added:
https://lobster-app-7doxp.ondigitalocean.app/ -> main app
https://lobster-app-7doxp.ondigitalocean.app/test -> Second app which gives errors if you look at console log.
https://lobster-app-7doxp.ondigitalocean.app/_nuxt/index.5c377f8b.js -> is the error
https://lobster-app-7doxp.ondigitalocean.app/test/_nuxt/index.5c377f8b.js -> works as expected

What does need to change in this https://gitlab.com/NLDylan/test repo so that these errors go away?
If you need more info then id happily share.

**Reply by danielroe** [reply]
You should set `app.baseURL` to `/test`: https://gitlab.com/NLDylan/test/-/blob/master/nuxt.config.ts

**Reply by NLDylan** [reply]
@danielroe 

import { defineNuxtConfig } from 'nuxt'

// https://v3.nuxtjs.org/api/configuration/nuxt.config
export default defineNuxtConfig({
    app: {
        baseURL: '/test'
    }
})

https://lobster-app-7doxp.ondigitalocean.app/test

If you look now, you see that the url is okay, but the application is not working anymore and results in 404.

Any suggestions what i can try or am doing wrong?

**Reply by danielroe** [reply]
Difficult to tell. The test app appears to be linking to the right files and at the right path. The real question is how you are _serving_ this `/test` endpoint. _Are_ the files present there?

**Reply by NLDylan** [reply]
@danielroe Yeah its kinda mind boggling at the moment. Well if i do no config at all my container just listens to /test and displays the app but with errors shown above.
If i change the config like you said it gives 404 errors, if i check the container in the /_nuxt folder i see thats everything there.
From what i can google i see that more people have run in to this problem and want to host their app as a sort of microservice on the same domain. Currently very ill so cant troubleshoot further..
